centreon-certified/signl4/signl4-events-apiv2.lua

```diff
@@ -0,0 +1,323 @@
+
+    local categories = self.sc_params.params.bbdo.categories
+    local elements = self.sc_params.params.bbdo.elements
+
+    self.format_event = {
+      [categories.neb.id] = {
+        [elements.host_status.id] = function () return self:format_event_host() end,
+        [elements.service_status.id] = function () return self:format_event_service() end
+      },
+      [categories.bam.id] = {}
+    }
+
+    self.send_data_method = {
+      [1] = function (data, element) return self:send_data(data, element) end
+    }
+
+    self.state_to_signlstatus_mapping = {
+        [0] = "resolved",
+        [1] = "new",
+        [2] = "new",
+        [3] = "new"
+    }
+
+    -- return EventQueue object
+    setmetatable(self, { __index = EventQueue })
+    return self
+  end
+
+--------------------------------------------------------------------------------
+---- EventQueue:format_event method
+---------------------------------------------------------------------------------
+function EventQueue:format_accepted_event()
+    local category = self.sc_event.event.category
+    local element = self.sc_event.event.element
+    local template = self.sc_params.params.format_template[category][element]
+    self.sc_logger:debug("[EventQueue:format_event]: starting format event")
+    self.sc_event.event.formated_event = {}
+
+    if self.format_template and template ~= nil and template ~= "" then
+      self.sc_event.event.formated_event = self.sc_macros:replace_sc_macro(template, self.sc_event.event, true)
+    else
+      -- can't format event if stream connector is not handling this kind of event and that it is not handled with a template file
+      if not self.format_event[category][element] then
+        self.sc_logger:error("[format_event]: You are trying to format an event with category: "
+          .. tostring(self.sc_params.params.reverse_category_mapping[category]) .. " and element: "
+          .. tostring(self.sc_params.params.reverse_element_mapping[category][element])
+          .. ". If it is a not a misconfiguration, you should create a format file to handle this kind of element")
+      else
+        self.format_event[category][element]()
+      end
+    end
+
+    self:add()
+    self.sc_logger:debug("[EventQueue:format_event]: event formatting is finished")
+  end
+
+  -- {
+  --   "Event Type": "HOST",
+  --   "Date": "HUMAN READABLE DATE TIME",
+  --   "Host": "Highway",
+  --   "Message": "to hell!",
+  --   "Status": "DOWN",
+  --   "State": "1",
+  --   "State Type": "1",
+  --   "Timestamp": "163[...]542"
+  --   "X-S4-Service": "Centreon",
+  --   "X-S4-AlertingScenario": "multi_ack", #NOT INCLUDED ATM
+  --   "X-S4-ExternalID": "HOSTALERT_666",
+  --   "X-S4-Status": "new"
+  -- }
+
+  function EventQueue:format_event_host()
+    self.sc_event.event.formated_event = {
+      "Event Type" = "HOST",
+      "Date" = self.sc_macros:replace_sc_macro(self.sc_event.event.last_check),
```

Comment:
do you mean 
https://github.com/centreon/centreon-stream-connector-scripts/blob/master/modules/docs/sc_macros.md#transform_date-method ?